### PR TITLE
Generator executor

### PR DESCRIPTION
**Pull request by wimax-grapl**
### Which issue does this PR correspond to?
Satisfies the first 2 checkboxes of 
https://github.com/grapl-security/issue-tracker/issues/733
https://app.zenhub.com/workspaces/grapl-6036cbd36bacff000ef314f2/issues/grapl-security/issue-tracker/733

My next steps are to do a couple of the blockers on this. 
(If it seems like I've been doing things out-of-order: I have been!) 

### What changes does this PR make to Grapl? Why?
Implement the Graph Generator Executor as described in the Plugins RFC.
https://github.com/grapl-security/grapl-rfcs/blob/main/text/0000-plugins.md#-executor-apis-graph-generator-executor-analyzer-executor

Still has quite a bit TODO.

### How were these changes tested?
So, here's the thing - this service is basically a one-file glue service. I'm not convinced it's worth unit testing, though I absolutely am obviously convinced it's worth integration testing.

However! 

As we've seen from previously introducing new services, you generally want to introduce the new docker image *before* you integrate it into Pulumi and our whole stack.

### CODE REVIEW

File: src/rust/plugin-executor/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+[dependencies]
+async-trait = "0.1.51"
+generator-sdk = { path = "../plugin-sdk/generator-sdk", features = ["client"] }
+plugin-work-queue = { path = "../plugin-work-queue" }
+rust-proto = { path = "../rust-proto" }
+structopt = { version = "0.3", default-features = false }
+tokio = { version = "1.17", features = ["macros", "rt", "rt-multi-thread"] }
+tonic = "0.6" # TODO move to 0.7 once we upgrade plugin-work-queue to -new
```
Comment (wimax-grapl) [inline src/rust/plugin-executor/Cargo.toml:25]: or just, yknow, remove it

**Review by jgrillo-grapl — COMMENTED** [review]
Would be 👍 👍 to port this to `rust-proto-new`

**Comment by wimax-grapl** [comment]
> Would be 👍 👍 to port this to `rust-proto-new`

Yep, but I'm gonna do that as a separate PR. Logic first, then upgrade.

File: src/rust/Dockerfile [hunk 2]
```diff
@@ -425,6 +425,15 @@ FROM rust-dist AS osquery-generator-deploy
 COPY --from=build /outputs/osquery-generator /
 ENTRYPOINT ["/osquery-generator"]
 
+##### generator-executor
+FROM rust-dist AS generator-executor-deploy
```
Comment (jgrillo-grapl) [inline src/rust/Dockerfile:429]: should this be "plugin-executor-deploy"?

**Review by jgrillo-grapl — APPROVED** [review]

File: src/rust/Dockerfile (hunk 2 above)
Comment (wimax-grapl) [reply src/rust/Dockerfile:429]: No, although this is up for reexamination later. If you look at Cargo.toml this package outputs 2 bins. The hope is to share logic between the two in this one package's lib.

**Pull request merged**